pkg/log/log.go

```diff
@@ -108,3 +112,38 @@ func parseLogLevel(logLevel string) (level.Option, error) {
 		return nil, fmt.Errorf("unrecognized log level %q", logLevel)
 	}
 }
+
+const logOnceTimedDuration = time.Second * 10
+
+// timedLogger logs from logStore every logOnceTimedDuration. It deletes the log entry from the store
+// after it has logged once.
+func timedLogger() {
+	t := time.NewTicker(logOnceTimedDuration)
+	for range t.C {
+		logStoreMux.Lock()
+		for k, logMsg := range logStore {
+			Debug("msg", "timed-debug")
+			Debug(logMsg...)
```

Comment:
```suggestion
			Debug("kind", "rate-limited", logMsg...)
```

Reply:
That will lead to compilation errors. However, I will implement a work-around here.